ego/bundle/main.go

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Edgeless Systems GmbH.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//go:build linux
+// +build linux
+
+package main
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"debug/elf"
+	"io"
+	"os"
+	"os/exec"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	"ego/cli"
+
+	"github.com/spf13/afero"
+)
+
+func main() {
+	// Open the current execeutable
+	execPath, err := os.Executable()
+	if err != nil {
+		panic(err)
+	}
+
+	// Parse ourselves as an ELF file so we get access to ELF sections later
+	elfFile, err := elf.Open(execPath)
+	if err != nil {
+		panic(err)
+	}
+	defer elfFile.Close()
+
+	// Use the "host" environment and not a mocked one for unit tests
+	fs := afero.NewOsFs()
+	runner := osRunner{}
+
+	// Run the unpacking & launch the unpacked enclave.
+	// Take the error from the enclave as the exit code, unless we failed to execute the enclave before.
+	exitCode, err := run(fs, elfFile, runner)
+	if err != nil {
+		panic(err)
+	} else {
+		os.Exit(exitCode)
+	}
+}
+
+func run(fs afero.Fs, selfElfFile *elf.File, runner cli.Runner) (int, error) {
+	// Create a temporary directory for both runtime and enclave
+	tempEGoRootPath, err := afero.TempDir(fs, "", "ego-bundle")
+	if err != nil {
+		return 1, err
+	}
+	defer fs.RemoveAll(tempEGoRootPath)
+
+	// Register cleanup handler to clean-up on STRG+C
+	cleanupHandler(tempEGoRootPath)
+
+	// Unpack the runtime and enclave
+	if err := unpackBundledImage(fs, tempEGoRootPath, selfElfFile); err != nil {
+		return 1, err
+	}
+
+	// Run the enclave
+	if exitCode, err := runEnclave(tempEGoRootPath, filepath.Join(tempEGoRootPath, "enclave"), runner); err != nil {
+		return 1, err
+	} else {
+		fs.RemoveAll(tempEGoRootPath)
+		return exitCode, nil
+	}
```

Comment:
```suggestion
	exitCode, err := runEnclave(tempEGoRootPath, filepath.Join(tempEGoRootPath, "enclave"), runner)
	if err != nil {
		return 1, err
	}

	return exitCode, nil
```
https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

RemoveAll is redundant because of the defer